### PR TITLE
Hooks Infra and New changes as per the UI requirement

### PR DESCRIPTION
**Pull request by kpunwatk**
Added Updated New Hook Infra for running QE UI hook Tests

### CODE REVIEW

**Review by mturley — APPROVED** [review]
LGTM. I'll wait for @ibragins to review before merging.

**Comment by mturley** [comment]
@kpunwatk I've converted this to a draft for now, when you're ready you can click the "Ready for review" button.

**Comment by kpunwatk** [comment]
> @kpunwatk I've converted this to a draft for now, when you're ready you can click the "Ready for review" button.

@mturley Thank You !

**Comment by kpunwatk** [comment]
@mturley Thank You !

File: qe-tests/cypress/integration/tests/vmware/config_separate_mapping.ts [hunk 1]
```diff
@@ -58,19 +65,32 @@ export const storageMapping: MappingData = {
...
+};*/
+
 export const planData: PlanData = {
   name: 'testplan-separate-mapping-cold',
   sProvider: providerData.name,
   tProvider: 'host',
   namespace: 'default',
-  sourceClusterName: 'Cluster',
-  vmwareSourceVmList: ['v2v-rhel7-igor'],
+  //sourceClusterName: 'Cluster',
```
Comment (ibragins) [inline qe-tests/cypress/integration/tests/vmware/config_separate_mapping.ts:83]: Please clean all comments of not-required code

File: qe-tests/cypress/integration/types/constants.ts [hunk 2]
```diff
@@ -31,10 +31,15 @@ export const finish = 'Finish';
 export const planSuccessMessage = 'Succeeded';
 export const planCanceledMessage = 'Canceled';
 export const SEC = 1000;
-
 export const CreateNewNetworkMapping = 'Create a network mapping';
 export const CreateNewStorageMapping = 'Create a storage mapping';
+//export const ansiblePlaybook = 'Ansible Playbook';
+//export const image = 'Custom container image';
```
Comment (ibragins) [inline qe-tests/cypress/integration/types/constants.ts:37]: Please clean all comments of not-required code

File: qe-tests/cypress/integration/tests/Rhv/config_separate_mapping_rhv.ts [hunk 3]
```diff
@@ -14,6 +15,8 @@ const v2v_rhv_username = Cypress.env('v2v_rhv_username');
 const v2v_rhv_password = Cypress.env('v2v_rhv_password');
 const v2v_rhv_hostname = Cypress.env('v2v_rhv_hostname');
 const v2v_rhv_cert = Cypress.env('v2v_rhv_cert');
+const preAnsiblePlaybook = Cypress.env('ansible_playbook');
+//const postAnsiblePlaybook = Cypress.env('ansible_playbook_1');
```
Comment (ibragins) [inline qe-tests/cypress/integration/tests/Rhv/config_separate_mapping_rhv.ts:19]: Please clean all comments of not-required code

File: qe-tests/cypress/integration/tests/Rhv/config_separate_mapping_rhv.ts [hunk 4]
```diff
@@ -66,19 +68,31 @@ export const storageMapping: MappingData = {
...
 
+export const preHookData: HookData = {
+  ansiblePlaybook: preAnsiblePlaybook,
+  //image: preImage,
+};
+
+/*export const postHookData: HookData = {
+  ansiblePlaybook: postAnsiblePlaybook,
+  //image: postImage,
+};*/
```
Comment (ibragins) [inline qe-tests/cypress/integration/tests/Rhv/config_separate_mapping_rhv.ts:79]: Please clean all comments of not-required code

File: qe-tests/cypress/integration/tests/ocpvirt/ocpVirtConfig.ts [hunk 5]
```diff
@@ -11,6 +11,7 @@ import { storageType } from '../../types/constants';
 const url = Cypress.env('url');
 const user_login = 'kubeadmin';
 const user_password = Cypress.env('pass');
+const ansible_playbook = Cypress.env('ansible_playbook');
```
Comment (ibragins) [inline qe-tests/cypress/integration/tests/ocpvirt/ocpVirtConfig.ts:14]: Please clean all variables that are not in use

File: qe-tests/cypress/integration/tests/vmware/config_separate_mapping.ts [hunk 6]
```diff
@@ -14,6 +15,8 @@ const v2v_vmware_username = Cypress.env('v2v_vmware_username');
 const v2v_vmware_password = Cypress.env('v2v_vmware_password');
 const v2v_vmware_hostname = Cypress.env('v2v_vmware_hostname');
 const v2v_vmware_cert = Cypress.env('v2v_vmware_cert');
+const preAnsiblePlaybook = Cypress.env('ansiblePlaybook');
+const postAnsiblePlaybook = Cypress.env('ansiblePlaybook_1');
```
Comment (ibragins) [inline qe-tests/cypress/integration/tests/vmware/config_separate_mapping.ts:19]: env variable name `ansiblePlaybook_1` is not acceptable, please update it to something like `postAnsiblePlaybook`

File: qe-tests/cypress/integration/tests/vmware/config_separate_mapping.ts [hunk 7]
```diff
@@ -58,19 +65,32 @@ export const storageMapping: MappingData = {
...
 
+export const preHookData: HookData = {
+  ansiblePlaybook: preAnsiblePlaybook,
+  //image: preImage,
+};
+
+/*export const postHookData: HookData = {
+  ansiblePlaybook: postAnsiblePlaybook,
+  //image: postImage,
+};*/
```
Comment (ibragins) [inline qe-tests/cypress/integration/tests/vmware/config_separate_mapping.ts:76]: Please clean all comments of not-required code

File: qe-tests/cypress/integration/tests/vmware/config_separate_mapping.ts (hunk 1 above)
Comment (ibragins) [inline qe-tests/cypress/integration/tests/vmware/config_separate_mapping.ts:83]: Please clean all comments of not-required code

File: qe-tests/cypress/integration/types/constants.ts (hunk 2 above)
Comment (ibragins) [inline qe-tests/cypress/integration/types/constants.ts:37]: Please clean all comments of not-required code

File: qe-tests/cypress/integration/types/constants.ts [hunk 8]
```diff
@@ -57,3 +62,9 @@ export enum summaryTitle {
   tProvider = 'Target provider',
   tNamespace = 'Target namespace',
 }
+
+export enum hookType {
+  prehook = 'Pre-migration',
+  posthook = 'Post-migration',
+}
+//export hookType = 'hook';
```
Comment (ibragins) [inline qe-tests/cypress/integration/types/constants.ts:70]: Please clean all comments of not-required code

File: qe-tests/cypress/integration/tests/vmware/config_separate_mapping.ts (hunk 1 above)
Comment (kpunwatk) [reply qe-tests/cypress/integration/tests/vmware/config_separate_mapping.ts:83]: Thanks for reviewing, I'll will clean up the not-required code.

**Review by mturley — APPROVED** [review]
LGTM with an initial pass, merging to unblock @ibragins . Will take a closer look after the holiday.

**Pull request merged**